### DISCUSSION TITLE
New features

### DISCUSSION THREAD

**Discussion by Alexandre-Fernandez** [Alexandre-Fernandez/astro-i18n#63] [Ideas]
If you want to help out developing new features here's what's planned :

- Support for [`root`](https://docs.astro.build/en/reference/configuration-reference/#root).
- Inherit options from the astro config as default values (`root`, `srcDir` and `trailingSlash`).
- Replace the `run` option in the config with a `serialization` option, values could be `"none"`, `"page"`, `"minimal"`. `"none"` would be the same as `run: "server"`, `"page"` would be the same as `run: "client+server"` and `"minimal"` would only serialize what has been used for the current page (see #62 ).
- An improvement on the last point would be to create an API that the client can call to fetch more translations in case a missing translation is loaded on the client-side.
- Add the ability to fetch a translation from a particular group in the `t` options, for example `t("key", undefined, { group: "common" })`. The type generation should give auto-completion for your available groups. Maybe a special string such as `"page"` could be used to reference the current page more easily.
- Integrate the [i18n routing changes from Astro](https://github.com/withastro/roadmap/pull/734) in `astro-i18n`.

This post will be edited to add new features or delete the ones that have been developed. 
Nothing is set in stone so we can discuss the implementation if you think that you have a better idea.
If you want to add another feature please contact me first, you can reply here or open a new discussion.

**Comment by gislerro** [comment]
Since Astro is getting baked in support for i18n based routing ([see this RFC](https://github.com/withastro/roadmap/pull/734)). I think it'll be great to integrate the changes into this library. As I understood it would make the generate pages script obsolete.

**Reply by Alexandre-Fernandez** [reply]
Unless I missed something ([RFC](https://github.com/withastro/roadmap/blob/i18n-routing/proposals/0041-i18n-routing.md)) it looks like the new astro i18n routing is just used to get the current localized route (including (sub)domains, etc) ?
From what I understand you still need to have your pages in the directory, however it would be cool to add some kind of watch command for the `sync` command.
We'll see exactly how it works when it's implemented.
As for now I do not have much time to add new features... still I added this one to the list.

**Comment by gislerro** [comment]
I've also been cooking up support for localized content/collections (.md/.mdx files) since translating blog/news posts via json is awkward. I'd be willing to contribute this functionality once I'm done on my current project (probably in 1-2 months). I think it would make this library a more complete solution to astro localization

**Reply by Alexandre-Fernandez** [reply]
Great, once you're ready to tackle this please tell me about the API you envision !

**Comment by Alexandre-Fernandez** [comment]
Support for [`srcDir`](https://docs.astro.build/en/reference/configuration-reference/#srcdir) in the `astro-i18n` config added.
Added type generation for every interpolation in a translation instead of just the first one (#67).
